core/chain-configs/README.md

```diff
@@ -0,0 +1,13 @@
+# Chain configs crate
+
+This crate provides typed interfaces to the NEAR Genesis and Client Configs, together with the functions to validate their correctness.
+
+## Genesis config
+Genesis config is the one that 'defines' the chain. It was set at the beginning and generally is not mutable. 
+
+## Client config
+
+Client config is the part of the config that client can configure on their own - it controls things like: how many peers it should connect to before syncing, which shards to track etc.
+
+## Protocol config
+This is the class that is spanning over GenesisConfig and RuntimeConfig. People should not use it directly, but use the ProtocolConfigView class instead.
```

Comment:
```suggestion
This is the type that is spanning over GenesisConfig and RuntimeConfig. People should not use it directly, but use the ProtocolConfigView class instead.
```

I generally call things types or data types. A term of art used in Rust compilers is `ADT`, which is a short for "A Data Type" :)

Reply:
@matklad wait I thought ADT stands for Algebraic Data Type?

Reply:
Yeah, the official correct definition for ADT is algebraic data types. 

In rust though, user-defined types are not, strictly speaking, ADTs: there are `unions`, product types come in two flavors (tuple and record structs), and recursive types are not possible without indirection. That's why calling some random Rust type ADT is pedantically wrong. The hack is to re-define "ADT" as an "A data type" :)

Reply:
fixed.